rita/src/middleware.rs

```diff
@@ -20,10 +26,27 @@ impl<S> Middleware<S> for Headers {
         if req.method() == &Method::OPTIONS {
             *resp.status_mut() = StatusCode::OK;
         }
-        resp.headers_mut().insert(
-            header::HeaderName::try_from("Access-Control-Allow-Origin").unwrap(),
-            header::HeaderValue::from_str(&format!("http://{}", url_no_port)).unwrap(),
-        );
+
+        let (key, cert) = {
+            let settings = SETTING.get_network();
+            (settings.key_path.clone(), settings.cert_path.clone())
+        };
+
+        if key.is_some()
```

Comment:
well this if statement will work by virtue of crashing if key or cert are none. Simple check to add